eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1456,7 +1461,7 @@ protected function internalPublishVersion(APIVersionInfo $versionInfo, $publicat
 
         $metadataUpdateStruct = new SPIMetadataUpdateStruct();
         $metadataUpdateStruct->publicationDate = isset($publicationDate) ? $publicationDate : time();
```

Comment:
@andrerom would something like this work:
```
if ($publicationDate === null && $versionInfo->getContentInfo()->publishedDate === null) {
    $publicationDate = time();
}
```

From what I gather, the published date is never set to timestamp 0, but to null (when I tried your snippet, got exception, so I tried to dig around a bit).

Alternatively, we could be a bit more explicit (while preserving the desired behaviour), and do something like this:
```
if ($publicationDate === null && $versionInfo->versionNo === 1) {
    $publicationDate = time();
}
```
This way it would be obvious from the code that we are setting the publication date to current time only if it is the first version, and publicationDate has not been set manually?